java/google/registry/rde/imports/RdeHostLinkAction.java

```diff
@@ -117,10 +117,18 @@ public void map(JaxbFragment<XjcRdeHostElement> fragment) {
         ofy().transact(new VoidWork() {
           @Override
           public void vrun() {
+            // link host to superordinate domain and set time of last superordinate change to
+            // the time of the import
             HostResource host =
                 ofy().load().now(Key.create(HostResource.class, xjcHost.getRoid()));
```

Comment:
From Nick:

This should handle the case where the HostResource can't be found.  This would occur in the common case where the host import MR skipped a host because it already exists, which is quite likely for common external hosts across TLDs (think ns1.googledomains.com or something).  In that case, this code will fail with an NPE.

There should be a test for this case.

I'm guessing it would be best to just skip these cases where the host isn't there, but we should probably increment a counter or something to track it.  It might be good to do a fallback load by foreign key in that case, and fail if no host exists with the FK at all (which more likely means something like someone forgot to run the host import MR before this one).  Note that we shouldn't just switch to loading by foreign key and still doing the mutations, since then we'd be setting lastSuperordinateChange to the import time incorrectly for existing hosts.